src/ceph-disk/ceph_disk/main.py

```diff
@@ -122,7 +122,7 @@ def get_ready_by_type(what):
 
     @staticmethod
     def get_ready_by_name(name):
-        return [x[name]['ready'] for x in PTYPE.values()]
+        return [x[name]['ready'] for x in PTYPE.values() if x.get(name, None)]
```

Comment:
@tchaikov I've one question about this:

The original code:
```
2536     def set_data_partition(self):
2537         if is_partition(self.args.data):
2538             LOG.debug('OSD data device %s is a partition',
2539                       self.args.data)
2540             self.partition = DevicePartition.factory(
2541                 path=None, dev=self.args.data, args=self.args)
2542             ptype = partition.get_ptype()
2543             ready = Ptype.get_ready_by_type('osd')
2544             if ptype not in ready:
2545                 LOG.warning('incorrect partition UUID: %s, expected %s'
2546                             % (ptype, str(ready)))
```

```
2238     def set_or_create_partition(self):
2239         if is_partition(self.args.lockbox):
2240             LOG.debug('OSD lockbox device %s is a partition',
2241                       self.args.lockbox)
2242             self.partition = DevicePartition.factory(
2243                 path=None, dev=self.args.lockbox, args=self.args)
2244             ptype = partition.get_ptype()
2245             ready = Ptype.get_ready_by_type('lockbox')
2246             if ptype not in ready:
2247                 LOG.warning('incorrect partition UUID: %s, expected %s'
2248                             % (ptype, str(ready)))
```

I think for data protection purpose, if find `if ptype not in ready:`, here should raise error `raise Error(xxxx)`.  If the user really wants to use this disk as ceph journal/data. We may add LOG.warning mentions how to setup the ceph partition. Such as:

```
/sbin/sgdisk --new=2:0:+5120M --change-name=2:ceph journal --partition-guid=2:81accb1a-8650-42f8-9534-1888c47b3090 --typecode=2:45b0969e-9b03-4f30-b4c6-b4b80ceff106 --mbrtogpt -- /dev/sdx
/sbin/sgdisk --largest-new=1 --change-name=1:ceph data --partition-guid=1:920ec801-433d-47fc-9872-c7449eb1901d --typecode=1:89c57f98-2fe5-4dc0-89c1-f3ad0ceff2be --mbrtogpt -- /dev/sdx
```

etc. or given some doc links.